src/Model.jl

```diff
@@ -94,24 +103,38 @@ function Model(lattice::AbstractMatrix{T},
                atoms::Vector{<:Element}=Element[],
                positions::Vector{<:AbstractVector}=Vec3{T}[];
                model_name="custom",
-               n_electrons::Int=sum(n_elec_valence, atoms; init=0),
+               n_electrons::Union{Int,Nothing}=nothing,
                magnetic_moments=T[],
                terms=[Kinetic()],
                temperature=zero(T),
                smearing=nothing,
                spin_polarization=default_spin_polarization(magnetic_moments),
                symmetries=default_symmetries(lattice, atoms, positions, magnetic_moments,
                                              spin_polarization, terms),
+               fixed_εF=nothing,
+               skip_fixed_εF_sane_check=false,
                ) where {T <: Real}
+
+    if !isnothing(fixed_εF)
+        # `n_electrons` incompatible with `fixed_εF`
+        @assert isnothing(n_electrons)
+        # Coulomb electrostratics does not make sense in the grand-canonical ensemble.
+        skip_fixed_εF_sane_check || @assert minmax(charge_nuclear.(atoms)...) == (0, 0)
+    elseif isnothing(n_electrons)
+        # Default value if not provided
+        n_electrons = sum(n_elec_valence, atoms; init=0)
+    elseif n_electrons < 0
+        error("`n_electrons` should be non-negative. Ensure to provide a " *
```

Comment:
you can just `@assert n_electrons >= 0`, if people pass a negative electron number they can debug that themselves

Reply:
Well you can always argue one or the other. I think this is very user facing and users deserve a nice message. Novices do all sort of crazy things (e.g. they might assume `n_electrons` behaves like a relative charge). It does not cost a lot of code and is just way nicer than an asserition.